test/test.c

```diff
@@ -33190,38 +33115,43 @@ test_MicroServiceStopsWhenServerStops(void)
     test("Connect for microservice: ");
     testCond(NATS_OK == natsConnection_Connect(&nc, opts));
 
-    test("Start microservice: ");
-    testCond(NULL == micro_AddService(&m, nc, &cfg));
-
+    _startMicroservice(&m, nc, &cfg, &arg);
+    
     test("Test microservice is running: ");
     testCond(!microService_IsStopped(m))
 
     test("Stop the server: ");
     testCond((_stopServer(serverPid), true));
 
+    nats_Sleep(1000);
```

Comment:
Not sure you need the sleep here and that may be the reason why you think that this takes longer than needed.

Reply:
🤦